Cue/app/reducers/library.js

```diff
@@ -39,16 +39,16 @@ function library(state: State = initialState, action: Action): State {
     }
 
   } else if (action.type == 'DECK_EDITED') {
-    let editedDeck = action.deck
+    let change = action.change
     decks.find( (deck, i) => {
-      if (deck.uuid == editedDeck.uuid)
-        return decks[i] = { ...deck, editedDeck }
+      if (deck.uuid == change.uuid)
+        return decks[i] = { ...deck, change }
     })
     if (!localChanges.find( (deck, i) => {
-      if (deck.uuid == editedDeck.uuid)
-        return localChanges[i] = { ...deck, editedDeck }
+      if (deck.uuid == change.uuid)
+        return localChanges[i] = { ...deck, change, action: deck.action }
```

Comment:
Sorry for all the back and forth... What happens if the `action` in `localChanges` is an `add` action? If it gets replaced with an `edit` action, wouldn't that cause problems when we try to sync? I think we'll need to do some smarter merging in the future to handle these cases.